### PR TITLE
Zen tweaks

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Add new package

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/cfs-zen-tweaks.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    maintainers = with maintainers; [ mkg20001 };
+  };
+
+  options = {
+    programs.cfs-zen-tweaks.enable = mkEnableOption "CFS Zen Tweaks";
+  };
+
+  config = mkIf cfg.enable {
+    systemd.packages = [ pkgs.cfs-zen-tweaks ];
+  };
```
Comment (aanderse) [inline nixos/modules/programs/cfs-zen-tweaks.nix:25]:
```suggestion
  config = mkIf cfg.enable {
    systemd.packages = [ pkgs.cfs-zen-tweaks ];

    systemd.services.set-cfs-tweak.wantedBy = [ "multi-user.target" "suspend.target" "hibernate.target" "hybrid-sleep.target" "suspend-then-hibernate.target" ];
  };
```

https://github.com/NixOS/nixpkgs/issues/81138

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/cfs-zen-tweaks/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  };
+
+  postPatch = ''
+    patchShebangs .
+  '';
+
+  postInstall = ''
+    chmod +x $out/lib/cfs-zen-tweaks/set-cfs-zen-tweaks.bash
+    wrapProgram $out/lib/cfs-zen-tweaks/set-cfs-zen-tweaks.bash \
+      --prefix PATH : ${gawk}/bin
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/cfs-zen-tweaks/default.nix:27]: It is only used here https://github.com/igo95862/cfs-zen-tweaks/blob/93b8cbac971d9d6cb1038716d05053f85f5d6e9f/set-cfs-zen-tweaks.bash#L36 so we can also patch it which is usually nicer than wrapping.

File: pkgs/os-specific/linux/cfs-zen-tweaks/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  src = fetchFromGitHub {
+    owner = "igo95862";
+    repo = "cfs-zen-tweaks";
+    rev = version;
+    sha256 = "HRR2tdjNmWyrpbcMlihSdb/7g/tHma3YyXogQpRCVyo=";
+  };
+
+  postPatch = ''
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/cfs-zen-tweaks/default.nix:21]: Which files exactly?

File: nixos/modules/programs/cfs-zen-tweaks.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# CFS Zen Tweaks
+
+{ config, pkgs, lib, ... }:
+
+with lib;
```
Comment (SuperSandro2000) [inline nixos/modules/programs/cfs-zen-tweaks.nix:5]: Not using with lib; over the entire file is encouraged.

**Review by aanderse — APPROVED** [review]
The module LGTM 👍

File: nixos/modules/programs/cfs-zen-tweaks.nix (hunk 4 above)
Comment (mkg20001) [reply nixos/modules/programs/cfs-zen-tweaks.nix:5]: Why? Does it make a difference in evaluation time?

**Pull request merged**